.github/workflows/binder-on-pr.yml

```diff
@@ -1,30 +1,15 @@
-# Reference https://mybinder.readthedocs.io/en/latest/howto/gh-actions-badges.html
 name: Binder Badge
 on:
   pull_request_target:
     types: [opened]
 
-permissions:
-  pull-requests: write
-
-
 jobs:
   binder:
     runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
     steps:
-      - name: comment on PR with Binder link
-        uses: actions/github-script@v3
+      - uses: jupyterlab/maintainer-tools/.github/actions/binder-link@v1
         with:
-          github-token: ${{secrets.GITHUB_TOKEN}}
-          script: |
-            var PR_HEAD_USERREPO = process.env.PR_HEAD_USERREPO;
-            var PR_HEAD_REF = process.env.PR_HEAD_REF;
-            github.issues.createComment({
-              issue_number: context.issue.number,
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              body: `[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/${PR_HEAD_USERREPO}/${PR_HEAD_REF}?urlpath=lab) :point_left: Launch a Binder on branch _${PR_HEAD_USERREPO}/${PR_HEAD_REF}_`
-            })
-        env:
-          PR_HEAD_REF: ${{ github.event.pull_request.head.ref }}
-          PR_HEAD_USERREPO: ${{ github.event.pull_request.head.repo.full_name }}
+
+          github_token: ${{ secrets.github_token }}
```

Comment:
This change optimizes "binder-on-pr" workflow by reusing "binder-link@v1" action instead of having to specify script and env variables manually. Let me test and if binder does not build, we can remove this.

Reply:
Binder still does not work so I'm ok to remove this step completely.